### PR TITLE
[Feature] CX-3439 Correctly mark links and buttons

### PR DESCRIPTION
**Pull request by josokinas**
# Problem
Buttons and links within the SDK are not accessible due to both outline being removed and/or divs (or spans) used instead of buttons.

# Solution
Certain parts of the code were converted to `<button>`, outlines added back, labels controlled via the aria.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [n/a] Have new manual tests been written down?
- [x] Have tests passed locally?
- [x] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://634-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/NavigationBar/index.js [hunk 1]
```diff
@@ -10,10 +10,16 @@ const NavigationBar = ({back, translate, disabled, isFullScreen, className}) =>
...
+      aria-label={translate('back')}
+      className={classNames(style.back, {
+        [style.disabled]: disabled,
+        [style.backHoverDesktop]: isDesktop
+      })}
+      onClick={preventDefaultOnClick(back)}
+    >
         <span className={style.iconBack} />
-        <span className={style.label}>
+        <span className={style.label} aria-hidden="true">
```
Comment (stefaniacardenas) [inline src/components/NavigationBar/index.js]: Why do you need `aria-hidden` here? Is it because the text disappears on small screens?

**Comment by stefaniacardenas** [comment]
Looks good! 👍 
Just two considerations.
* On the Document uploader screen there is an icon with the same function of a button. When you click on it, it triggers the file input just like the `Upload file` button. I wonder if we should consider that a button or not.
* When you click on `Enlarge image` button to inspect the document, it's difficult to focus on the Close button because the screen reader can read the elements underneath the popup. 
I think we need to create a ticket to make sure that the enlarged image pop up is announced and one to hide all the elements from screen readers once the popup is open.
What do you think @seewah ?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://634-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/NavigationBar/index.js (hunk 1 above)
Comment (josokinas) [reply src/components/NavigationBar/index.js]: That's correct. So the decision here was to hide the actual copy within the span and add `aria-label` to button itself that is read by screen reader all the time.

**Comment by josokinas** [comment]
> Looks good! 👍
> Just two considerations.
> 
> * On the Document uploader screen there is an icon with the same function of a button. When you click on it, it triggers the file input just like the `Upload file` button. I wonder if we should consider that a button or not.
> * When you click on `Enlarge image` button to inspect the document, it's difficult to focus on the Close button because the screen reader can read the elements underneath the popup.
>   I think we need to create a ticket to make sure that the enlarged image pop up is announced and one to hide all the elements from screen readers once the popup is open.
>   What do you think @seewah ?

**Comment by josokinas** [comment]
Closed accidentally.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by josokinas** [comment]
@stefaniacardenas In regards to your comments:
- Not sure if this is needed, because tabbing focuses on an actual button. With keyboard it's OK to have a clickable area, but with keyboard is this needed? In such case you would be tabbing twice for same action - once on an area, and then on a button.
- Agree with this, do you think additional ticket is needed, or this should be scoped with this PR?

**Comment by stefaniacardenas** [comment]
I think it can be done as part of another ticket

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-634-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://634-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**